app/controllers/api/templates_controller.rb

```diff
@@ -4,8 +4,51 @@ class TemplatesController < BaseController
     include Subcollections::PolicyProfiles
     include Subcollections::Tags
 
-    def edit_resource(type, id = nil, data = {})
-      super(type, id, data.extract!('name', 'description'))
+    RELATIONSHIP_COLLECTIONS = %w[vms templates].freeze
+    VALID_EDIT_ATTRS = %w[description custom_1 name child_resources parent_resource].freeze
+
+    def edit_resource(type, id, data)
+      attrs = validate_edit_data(data)
+      parent, children = build_parent_children(data)
+      resource_search(id, type).tap do |template|
+        template.replace_children(children)
+        template.set_parent(parent)
+        template.update!(attrs)
+      end
+    rescue => err
+      raise BadRequestError, "Cannot edit Template - #{err}"
+    end
+
+    def validate_edit_data(data)
+      invalid_keys = data.keys - VALID_EDIT_ATTRS
+      raise BadRequestError, "Cannot edit values #{invalid_keys.join(', ')}" if invalid_keys.present?
+
+      data.except('parent_resource', 'child_resources')
+    end
+
+    def build_parent_children(data)
+      children = if data.key?('child_resources')
+                   data['child_resources'].collect do |child|
+                     fetch_relationship(child['href'])
+                   end
+                 end
+
+      parent = if data.key?('parent_resource')
+                 if data['parent_resource'].nil?
+                   nil
+                 else
+                   fetch_relationship(data['parent_resource']['href'])
+                 end
+               end
+
+      [parent, Array(children)]
```

Comment:
I think you want to use `Array.wrap()` here:

```suggestion
      [parent, Array.wrap(children)]
```

not sure if you want `Array(children)` or `Array.wrap(children)`
If `children = [child]`, did you want `[[child]]` or `[child]` coming back?
Also note `Array.wrap(nil) # => nil`